Sources/Clickstream/ClickstreamAnalytics.swift

```diff
@@ -89,6 +90,30 @@ public enum ClickstreamAnalytics {
         Amplify.Analytics.enable()
     }
 
+    static func getAmplifyConfigurationSafely(_ bundle: Bundle = Bundle.main) -> AmplifyConfiguration {
+        var amplifyConfiguraion: AmplifyConfiguration!
+        do {
+            guard let path = bundle.path(forResource: "amplifyconfiguration", ofType: "json") else {
+                throw ConfigurationError.invalidAmplifyConfigurationFile(
+                    "Could not load default `amplifyconfiguration.json` file",
+                    "please check if you added the correct configuration file")
+            }
+            let url = URL(fileURLWithPath: path)
+            amplifyConfiguraion = try AmplifyConfiguration(configurationFile: url)
+        } catch {
+            log.debug("Could not load default `amplifyconfiguration.json` file")
```

Comment:
Why not move to the above `else` clause?

I think it's by design not a exception.

Reply:
Sure, I will simplify it in the `else` clause.